File: javascript/lib/GcpconnectorExample.js
/*
 * Copyright 2019 Trend Micro.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Create a GCP connector.
 * @param {ApiClient} api           The Deep Security API exports.
 * @param {String} name             The name of the GCP connector.
 * @param {String} serviceAccount   The GCP service account used by the GCP connector.
 * @param {String} apiVersion       The API version to use.
 * @returns {Promise}               The promise contains the GCP connector object.
 */
exports.createGcpConnector = function(api, name, serviceAccount, apiVersion){
    let apiInstance = new api.GCPConnectorsApi();
    let gcpConnector = new api.GCPConnector();
    // Set the GCP connector properties.
    gcpConnector.name = name;
    gcpConnector.serviceAccount = serviceAccount;
    return apiInstance.createGCPConnector(gcpConnector, apiVersion);
}

/**
 * Submit a synchronize action of a GCP connector by ID.
 * @param {ApiClient} api           The Deep Security API exports.
 * @param {Number} gcpConnectorID   The ID of the GCP connector.
 * @returns {Promise}               The promise contains GCP connector object.
 */
exports.submitGcpConnectorAction = function(api, gcpConnectorID, apiVersion){
    let apiInstance = new api.GCPConnectorActionsApi();
    // Set the action.
    let gcpConnectorAction = new api.Action();
    gcpConnectorAction.type = "synchronize";
    return apiInstance.createGCPConnectorAction(gcpConnectorID, gcpConnectorAction, apiVersion);
}

File: java/src/com/trendmicro/deepsecurity/docs/GcpConnectorExample.java
// Copyright (C) 2019 Trend Micro Inc. All rights reserved.

package com.trendmicro.deepsecurity.docs;

import com.trendmicro.deepsecurity.ApiException;
import com.trendmicro.deepsecurity.api.GcpConnectorActionsApi;
import com.trendmicro.deepsecurity.api.GcpConnectorsApi;
import com.trendmicro.deepsecurity.model.Action;
import com.trendmicro.deepsecurity.model.GCPConnector;

/**
 * Examples of the GCP connector. 
 */
public class GcpConnectorExample {

	/**
	 * Create a GCP connector.
	 * 
	 * @param name           	The name of the GCP connector.
	 * @param serviceAccount 	The base64 string of the GCP service account.
	 * @param apiVersion     	The version of the API to use.
	 * @return 			Return the GCP connector object.
	 * @throws ApiException 	Throw an exception if a problem occurs when creating a GCP connector on
	 *                      	Deep Security Manager.
	 */
	public static GCPConnector createGcpConnector(String name, String serviceAccount, String apiVersion)
			throws ApiException {

		GcpConnectorsApi apiInstance = new GcpConnectorsApi();

		// Create a GCP connector.
		GCPConnector gcpConnector = new GCPConnector();
		gcpConnector.setName(name);
		gcpConnector.setServiceAccount(serviceAccount);

		// Add a GCP connector to Deep Security Manager.
		GCPConnector result = null;
		result = apiInstance.createGCPConnector(gcpConnector, apiVersion);

		return result;
	}

	/**
	 * Submit a sync action to the GCP connector.
	 * @param gcpConnectorID 	The target gcpConnectorID .
	 * @param apiVersion 		The version of the API to use.
	 * @return 			Return the action object which contains the action status.
	 * @throws ApiException 	Throw an exception if a problem occurs when creating a GCP connector on
	 *                      	Deep Security Manager.
	 */
	public static Action submitGCPConnectorSyncAction(Integer gcpConnectorID, String apiVersion) throws ApiException {
		
		GcpConnectorActionsApi apiInstance = new GcpConnectorActionsApi();
		//Set up the synchronize action.
		Action gcpConnectorAction = new Action();
		gcpConnectorAction.setType("synchronize");
		Action result = apiInstance.createGCPConnectorAction(gcpConnectorID, gcpConnectorAction, apiVersion);
		return result;
	}

}